typing/ctype.ml

```diff
@@ -1652,10 +1652,14 @@ let expand_head_once env ty =
 (* Check whether a type can be expanded *)
 let safe_abbrev env ty =
   let snap = Btype.snapshot () in
-  try ignore (expand_abbrev env ty); true
-  with Cannot_expand | Unify _ ->
-    Btype.backtrack snap;
-    false
+  try ignore (expand_abbrev env ty); true with
+    Cannot_expand ->
+      Btype.backtrack snap;
+      false
+  | Unify _ ->
+      Btype.backtrack snap;
+      cleanup_abbrev ();
+      false
```

Comment:
Why did this change introduce a call to `cleanup_abbrev`? My understanding is that `Btype.backtrack` already calls `cleanup_abbrev`.